spec/support/request/web_helper.rb

```diff
@@ -152,6 +152,16 @@ def accept_js_alert
     page.driver.browser.switch_to.alert.accept
   end
 
+  def angular_http_requests_finished(angular_controller=nil)
+    scope_element = angular_controller ? "[ng-controller=#{angular_controller}]" : '.ng-scope'
+    page.evaluate_script("angular.element(document.querySelector('#{scope_element}')).injector().get('$http').pendingRequests.length == 0")
+  end
+
+  def request_monitor_finished(angular_controller=nil)
+    scope_element = angular_controller ? "[ng-controller=#{angular_controller}]" : '.ng-scope'
+    page.evaluate_script("angular.element(document.querySelector('#{scope_element}')).scope().RequestMonitor.loading == false")
+  end
+
```

Comment:
I think this will make it more readable:
```suggestion
.
def angular_http_requests_finished(angular_controller=nil)
    scope_expr = scope_element_expression(angular_controller)
    pending_http_requests_script = "#{scope_expr}.injector().get('$http').pendingRequests.length == 0"
    page.evaluate_script(pending_http_requests_script)
end

 def request_monitor_finished(angular_controller=nil)
    scope_expr = scope_element_expression(angular_controller)
    request_monitor_script = "#{scope_expr}.scope().RequestMonitor.loading == false")"
    page.evaluate_script(request_monitor_script)
  end
  
  def scope_element_expression(angular_controller=nil)
      scope_element = angular_controller ? "[ng-controller=#{angular_controller}]" : '.ng-scope'
      scope_element_expression = "angular.element(document.querySelector('#{scope_element}')"
  end
```

Reply:
I've tidied it up and added a comment to explain how the format for the angular controller name  and how it finds the scope if `nil` :+1: